### PR TITLE
Don't change heading on a unit that doesn't need it.

### PR DESCRIPTION
**Pull request by marcushutchings**
An optimization that had been previously applied, but with a fix that addresses unit not alwasy animating correctly when created.

### CODE REVIEW

File: rts/Sim/MoveTypes/GroundMoveType.cpp [hunk 1]
```diff
@@ -1385,11 +1387,16 @@ void CGroundMoveType::ChangeHeading(short newHeading) {
 	if (owner->GetTransporter() != nullptr)
 		return;
 
+	if (wantedHeading != newHeading)
+		wantedHeading = newHeading;
```
Comment (sprunk) [inline rts/Sim/MoveTypes/GroundMoveType.cpp:1391]:
```suggestion
	wantedHeading = newHeading;
```

File: rts/Sim/MoveTypes/GroundMoveType.cpp [hunk 2]
```diff
@@ -518,6 +518,8 @@ CGroundMoveType::CGroundMoveType(CUnit* owner):
 
 	forceStaticObjectCheck = true;
 
+	flatFrontDir = (owner->frontdir * XZVector).Normalize();
```
Comment (sprunk) [inline rts/Sim/MoveTypes/GroundMoveType.cpp:521]: looks unrelated?

File: rts/Sim/MoveTypes/GroundMoveType.cpp (hunk 1 above)
Comment (marcushutchings) [reply rts/Sim/MoveTypes/GroundMoveType.cpp:1391]: To check: does a non-changing store avoid a cache line write-back?

File: rts/Sim/MoveTypes/GroundMoveType.cpp (hunk 2 above)
Comment (marcushutchings) [reply rts/Sim/MoveTypes/GroundMoveType.cpp:521]: It is completely related. FlatFrontDir has to be initialized. If it isn't and a new unit tries to move without trying to turn, then it will not animate correctly and it will not move at the correct speed.

File: rts/Sim/MoveTypes/GroundMoveType.cpp (hunk 1 above)
Comment (marcushutchings) [reply rts/Sim/MoveTypes/GroundMoveType.cpp:1391]: Okay. Checking through the details. Intel does implement tracking of cache line state.
